File: protocol/resources/python/tv/searchAll.py
from __future__ import print_function
import os
import sys
sys.path.append(os.path.abspath(os.path.dirname(os.path.dirname(__file__))))
from aurum_pb2 import *
from aurum_pb2_grpc import BootstrapStub
import logging
import grpc
import time

# Find TextField and input "Movie" text
def SearchTestWithText(stub):
    response = stub.findElement(ReqFindElement(widgetType='TextField'))
    if len(response.elements) <= 0: return False
    targetObj = response.elements[0].elementId
    testString = 'Movie'
    stub.setValue(ReqSetValue(elementId=targetObj, stringValue=testString))

    # Wait until result upload
    time.sleep(2)

    return True

# Find Foused item and move focus to right then compare focused item with previous one
def SearchFocusedObject(stub):
    response = stub.findElement(ReqFindElement(isFocused=True))
    if len(response.elements) <= 0: return False

    prevObj = response.elements[0].elementId
    stub.sendKey(ReqKey(type='XF86', actionType='LONG_STROKE', XF86keyCode='Right'))
    time.sleep(1)

    response = stub.findElement(ReqFindElement(isFocused=True))
    if len(response.elements) <= 0: return False

    if prevObj != response.elements[0].elementId:
        return True

    return False


# Launch application. it returns application running state
def launchAppTest(stub):
    stub.sendKey(ReqKey(type='XF86', actionType='STROKE', XF86keyCode='XF86Search'))
    # Wait until app launch
    time.sleep(5)
    return stub.getAppInfo(ReqGetAppInfo(packageName='com.samsung.tv.searchall')).isRunning

# Close application. it returns application running state
def closeAppTest(stub):
    stub.sendKey(ReqKey(type='XF86', actionType='STROKE', XF86keyCode='XF86Exit'))
    time.sleep(2)
    return stub.getAppInfo(ReqGetAppInfo(packageName='com.samsung.tv.searchall')).isRunning != True

def runTest(stub, testFunc):
    print("Testing started :", testFunc)

    result = testFunc(stub)

    print("Testing result :", result)
    return True

def run():
    with grpc.insecure_channel('127.0.0.1:50051') as channel:
        stub = BootstrapStub(channel)
        runTest(stub, launchAppTest)
        runTest(stub, SearchTestWithText)
        runTest(stub, SearchFocusedObject)
        runTest(stub, closeAppTest)

if __name__ == '__main__':
    logging.basicConfig()
    run()

File: protocol/resources/node/tv/searchAll.js
import * as grpc from "@grpc/grpc-js";
import * as protoLoader from "@grpc/proto-loader";
import assert, { doesNotMatch } from "assert";

let PROTO_PATH = "./aurum.proto";
let packageDefinition = protoLoader.loadSync(PROTO_PATH, {
  keepCase: true,
  longs: String,
  enums: String,
  defaults: true,
  oneofs: true
});

let aurum = grpc.loadPackageDefinition(packageDefinition).aurum;

describe("Aurum Nodejs Test", function () {
  let client;
  let searchAll;
  let textField;

  before(() => {    
    client = new aurum.Bootstrap(
      "localhost:50051",
      grpc.credentials.createInsecure()
    );
  });
  
  describe("SearchAll Test", () => {

    it("Launch App", function (done) {
      this.timeout(15000);
      client.launchApp({ packageName: "com.samsung.tv.searchall" }, (err, response) => {
        if (err) {
          console.log(err);
          done(err);
          return;
        }
        else {
          assert.strictEqual(response.status, "OK");
          done();
          return;
        }
      });
    });

    /*it("Find SearchAll Window", function (done) {
      this.timeout(15000);
      client.findElement({ maxDepth: 1, packageName: "com.samsung.tv.searchall", isActive: true, isShowing: true }, (err, response) => {
        if (err) {
          console.log(err);
	        done(err);
        } else {
          let returnList = response.elements;
          assert.ok(returnList.length > 0);
          console.log(returnList[0]);
          searchAll = returnList[0];
          assert.strictEqual(searchAll.package, "com.samsung.tv.searchall");
          done();
        }
      });
  });*/

    it("Find TextField", function (done) {
      this.timeout(15000);
      // assert.notEqual(searchAll, undefined);

      client.findElement({ widgetType: "TextField", packageName: "com.samsung.tv.searchall" }, (err, response) => {
        if (err) {
          console.log(err);
	  done(err);
        } else {
          let textField = response.element;
          assert.null(textField);

          assert.strictEqual(textField.widget_type, "TextField");
          done();
        }
      });
    });

    it("Set Text", function (done) {
      this.timeout(15000);
      assert.notEqual(textField, undefined);
      client.setValue({ elementId: textField.elementId, type: 0, stringValue: "Movie" }, (err, response) => {
        if (err) {
          console.log(err);
          done(err);
        } else {
          assert.strictEqual(response.status, "OK");
          done();
        }
      });      
    })
  });
});
